src/main/java/com/conveyal/r5/streets/StreetRouter.java

```diff
@@ -317,11 +317,14 @@ public boolean setOrigin (double lat, double lon) {
         // Uses weight based on distance from end vertices, and speed on edge which depends on transport mode
         float speedMetersPerSecond = edge.calculateSpeed(profileRequest, streetMode);
         startState1.weight = (int) ((split.distance1_mm / 1000) / speedMetersPerSecond);
+        startState1.durationSeconds = startState1.weight;
+        startState1.distance = split.distance1_mm;
         edge.advance();
         // Speed can be different on opposite sides of the same street
         speedMetersPerSecond = edge.calculateSpeed(profileRequest, streetMode);
         startState0.weight = (int) ((split.distance0_mm / 1000) / speedMetersPerSecond);
-        // FIXME we're setting weight but not time and distance on these states above!
```

Comment:
Ack, looks like the kind of thing that should have just been resolved at the time the FIXME comment was added. But often those FIXMEs are added while in the midst of another change or bugfix.